HWKAGENT-158 Collection Interval Changes

jshaughn opened the pull request:
- add new subsystem configuration for minimum collection interval
  - default to 30s (changed from hardcode of 1s)
- let collection interval 0 indicate collection is disabled
- consolidate xml and Protocol service on 600s auto-discovery

@jmazzitelli You can take a look as is to see if it looks kosher.  I didn't add any tests yet, I'll look at that on Monday.

jmazzitelli commented:
LGTM

jshaughn commented:
@jmazzitelli Added an itest (and a little itest infra) but it's not working yet.  It's going to need the full agent restart stuff to work, triggered when updating a metric/avail attribute.  So, waiting on that PR and then I'll rebase and retest.

jshaughn commented:
@jmazzitelli I rebased on master and I've updated AgentInstallerITest to test for a disabled metric (and implicitly a collection interval update restarting the agent).  I think is ready for merge.

jmazzitelli commented on hawkular-wildfly-agent-itest-parent/hawkular-wildfly-agent-itest-util/src/main/java/org/hawkular/wildfly/agent/itest/util/AbstractITest.java:
```diff
@@ -222,6 +222,31 @@ protected String getNodeAttribute(ModelControllerClient mcc, ModelNode addressAc
+                .assertSuccess();
+    }
+
+    protected void restartAgent() {
```
do you still need this restartAgent() method? Get rid of it if not needed anymore.

jshaughn replied:
yeah, not called anymore, I'll remove.

jshaughn commented:
Hold On, checking one thing...

jshaughn committed:
HWKAGENT-158 Collection Interval Changes
- add new subsystem configuration for minimum collection interval
  - default to 30s (changed from hardcode of 1s)
- let collection interval 0 indicate collection is disabled
- consolidate xml and Protocol service on 600s auto-discovery
- Add an itest for a disabled metric collection (via interval=0)
  - fix invalid test condition on some existing tests, they
    were passing for the wrong reason.
- add hawkular.log.root to itest env to allow for logging changes
- provide writeNodeAttribute support in AbstractITest
- update some itest log props to match the standalone.xml
